docs/examples/application/expenses-spring-boot/src/main/java/com/example/springboot/Authorizer.java

```diff
@@ -0,0 +1,80 @@
+package com.example.springboot;
+
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import com.osohq.oso.Http;
+import com.osohq.oso.Oso;
+import com.osohq.oso.Exceptions.OsoException;
+
+@Component
+public class Authorizer extends HandlerInterceptorAdapter {
+    @Resource(name = "setupOso")
+    private Oso oso;
+
+    @Resource(name = "requestScopeCurrentUser")
+    private CurrentUser currentUser;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+        try {
+            setCurrentUser(request);
+
+            // Authorize the incoming request
+            Http http = new Http(request.getServerName(), request.getServletPath().toString(), getQuery(request));
+            if (!oso.isAllowed(currentUser.get(), request.getMethod(), http)) {
+                throw new ResponseStatusException(HttpStatus.FORBIDDEN, "oso authorization: unauthorized");
+            }
+        } catch (SQLException e) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "User not found", e);
```

Comment:
This one was actually okay as a `Unauthorized`/401 since that typically indicates not authenticated.